src/OrbitGl/App.cpp

```diff
@@ -2564,3 +2569,32 @@ void OrbitApp::CaptureMetricProcessApiTimer(const orbit_client_protos::TimerInfo
       UNREACHABLE();
   }
 }
+
+void OrbitApp::SaveCaptureUserInfo() {
+  CHECK(std::this_thread::get_id() == main_thread_id_);
+  if (!HasCaptureData()) return;
+  if (IsCapturing()) {
+    // We are going to save it at the end of capture anyways.
+    return;
+  }
+
+  const auto& file_path = GetCaptureData().file_path();
+  if (!file_path.has_value()) {
+    LOG("Warning: capture is not backed by a file, skipping the save.");
+    return;
+  }
+
+  orbit_client_protos::UserDefinedCaptureInfo capture_info;
+  const auto& frame_track_function_ids = GetCaptureData().frame_track_function_ids();
+  *capture_info.mutable_frame_tracks_info()->mutable_frame_track_function_ids() = {
+      frame_track_function_ids.begin(), frame_track_function_ids.end()};
+  thread_pool_->Schedule([this, capture_info = std::move(capture_info),
+                          file_path = file_path.value()] {
+    LOG("Saving user defined capture info to \"%s\"", file_path.string());
+    auto write_result = orbit_capture_file::WriteUserData(file_path, capture_info);
+    if (write_result.has_error()) {
+      SendErrorToUi("Save failed", absl::StrFormat("Save to \"%s\" failed: %s", file_path.string(),
```

Comment:
The following is basically a different idea how to structure this whole function. It requires significantly more code at call site and maybe thats a bad tradeoff. So feel free to not do this

This error is displayed to the user directly and in my opinion might be confusing to the user. So maybe `SaveCaptureUserInfo` can return `ErrorMessageOr<void>` and return this string here as an error message. Then the callsite could augment this error with more context (e.g. Unable to add frame track to the capture file on disk, error: Save to ... failed: ...)
But this also means the call site needs to decide whether to call `SaveCaptureUserInfo` and thats maybe uncomfortable. 

In case you dont do this, maybe also consider naming this function `TrySaveCaptureUserInfo`.

Reply:
> But this also means the call site needs to decide whether to call SaveCaptureUserInfo and thats maybe uncomfortable.

I do not think I got this part. Why does caller side need to decide?

Reply:
Also you probably meant `Future<ErrorMessageOr<void>>`

Reply:
and I will need to implement ScheduleAfterIfError()... I am going to leave this as it is for now.

Reply:
> I do not think I got this part. Why does caller side need to decide?

I just meant that most of the checks that you are doing in this function would need to be done at call site (for example `if (!file_path.has_value())`. But now it doesnt matter anyways ;)